apps/site/lib/site_web/controllers/schedule/line/diagram_helpers.ex

```diff
@@ -140,6 +140,11 @@ defmodule SiteWeb.ScheduleController.Line.DiagramHelpers do
 
   # Splits green branch into a tuple of shared stops and stops that are unique to that branch.
   @spec split_green_branch(RouteStops.t(), direction_id) :: {[RouteStop.t()], [RouteStop.t()]}
+
+  defp split_green_branch(%RouteStops{stops: []}, _direction_id) do
+    {[], []}
+  end
+
```

Comment:
Not sure if I'm getting my wires tangled, but I think `split_green_branch` gets called by `DiagramHelpers.build_stop_list()` on line 118 of line_api.ex.  But I think the Sentry error is getting triggered the line before, in `LineHelpers.get_branch_route_stops`.

The stack trace:
`branch(stops)` route_stops line 51
`Enum.map(route_stop_groups, &from_list/1)` route_stops line 21
`RouteStops.from_route_stop_groups()`, line/helpers line 65
`LineHelpers.get_branch_route_stops`, line_api line 117

So -- not _totally_ sure in elixir -- but I think the error will happen before `DiagramHelpers.build_stop_list()` is even executed.

I messaged back and forth with Gabe about this weird issue, and he thinks it might be a caching issue with dotcom.  (oh boy...)  The error always happens a few hours after a GTFS deployment.  Gabe hypothesized that "once we deploy a new version [of GTFS], the `If-Modified-Since` for a lot of the [GTFS] resources will be changing. maybe some of dotcom's requests time out, but over time the successful ones get cached, and then eventually it's steady state again".  And he pointed out this has mostly been an issue with the D line (because of all the shuttle weirdness, I think).

Ok so, wow.  Caching is a little out of my league at the moment.  My first idea was to just filter out empty arrays on line 63 of `LineHelpers.get_branch_route_stops`, because I think that's the root problem.  Somewhere in there, a request fails and an empty array is returned instead of the D line stops array.  But if we did that, then would we lose the D branch from the line diagram?!  Ha, that's no good either.  Sorry, comment is too long.  But I'm stumped for the day!